test/test-mongocrypt-ctx-encrypt.c

```diff
@@ -2163,6 +2163,373 @@ _test_encrypt_fle2_unindexed_encrypted_payload (_mongocrypt_tester_t *tester)
+   localkey = mongocrypt_binary_new_from_data ((uint8_t *) localkey_data,
+                                               sizeof localkey_data);
+   ASSERT_OK (mongocrypt_setopt_kms_provider_local (crypt, localkey), crypt);
+   ASSERT_OK (mongocrypt_setopt_crypto_hooks (
+                 crypt,
+                 _std_hook_native_crypto_aes_256_cbc_encrypt,
+                 _std_hook_native_crypto_aes_256_cbc_decrypt,
+                 _test_rng_source,
+                 _std_hook_native_hmac_sha512,
+                 _std_hook_native_hmac_sha256,
+                 _error_hook_native_sha256,
+                 rng_source /* ctx */),
+              crypt);
+
+   mongocrypt_binary_destroy (localkey);
+   ASSERT_OK (mongocrypt_init (crypt), crypt);
+   return crypt;
+}
+
+static void
+_test_encrypt_fle2_explicit (_mongocrypt_tester_t *tester)
+{
+   _mongocrypt_buffer_t user_key_id;
+   _mongocrypt_buffer_t index_key_id;
+
+   if (!_aes_ctr_is_supported_by_os) {
+      printf ("Common Crypto with no CTR support detected. Skipping.");
+      return;
+   }
+
+   _mongocrypt_buffer_copy_from_hex (&user_key_id,
+                                     "ABCDEFAB123498761234123456789012");
+   _mongocrypt_buffer_copy_from_hex (&index_key_id,
+                                     "12345678123498761234123456789012");
+
+   /* Test Unindexed. */
+   {
+#define RNG_DATA \
+   "\x4d\x06\x95\x64\xf5\xa0\x5e\x9e\x35\x23\xb9\x8f\x57\x5a\xcb\x15"
+      _test_rng_data_source source = {
+         .buf = {.data = (uint8_t *) RNG_DATA, .len = sizeof (RNG_DATA)}};
+#undef RNG_DATA
+      mongocrypt_t *crypt = _crypt_with_rng (&source);
+      mongocrypt_ctx_t *ctx = mongocrypt_ctx_new (crypt);
+
+      ASSERT_OK (
+         mongocrypt_ctx_setopt_index_type (ctx, MONGOCRYPT_INDEX_TYPE_NONE),
+         ctx);
+      ASSERT_OK (mongocrypt_ctx_setopt_key_id (
+                    ctx, _mongocrypt_buffer_as_binary (&user_key_id)),
+                 ctx);
+      ASSERT_OK (mongocrypt_ctx_setopt_index_key_id (
+                    ctx, _mongocrypt_buffer_as_binary (&index_key_id)),
+                 ctx);
+      ASSERT_OK (mongocrypt_ctx_explicit_encrypt_init (
+                    ctx, TEST_BSON ("{'v': 'value123'}")),
+                 ctx);
+
+      ASSERT_STATE_EQUAL (mongocrypt_ctx_state (ctx),
+                          MONGOCRYPT_CTX_NEED_MONGO_KEYS);
+      {
+         ASSERT_OK (mongocrypt_ctx_mongo_feed (
+                       ctx, TEST_FILE ("./test/data/keys/"
+                                       "12345678123498761234123456789012-local-"
+                                       "document.json")),
+                    ctx);
+         ASSERT_OK (mongocrypt_ctx_mongo_feed (
+                       ctx, TEST_FILE ("./test/data/keys/"
+                                       "ABCDEFAB123498761234123456789012-local-"
+                                       "document.json")),
+                    ctx);
```

Comment:
```suggestion
         ASSERT_OK (mongocrypt_ctx_mongo_feed (
                       ctx,
                       TEST_FILE ("./test/data/keys/"
                                  "12345678123498761234123456789012-local-"
                                  "document.json")),
                    ctx);
         ASSERT_OK (mongocrypt_ctx_mongo_feed (
                       ctx,
                       TEST_FILE ("./test/data/keys/"
                                  "ABCDEFAB123498761234123456789012-local-"
                                  "document.json")),
                    ctx);
```
Clang-Format. Also applies to other instances of loading `./test/data/keys/...` below.